lineapy/api/api.py

```diff
@@ -359,32 +377,88 @@ def to_pipeline(
     framework: str = "SCRIPT",
     pipeline_name: Optional[str] = None,
     dependencies: TaskGraphEdge = {},
-    pipeline_dag_config: Optional[AirflowDagConfig] = {},
     output_dir: str = ".",
+    input_parameters: List[str] = [],
+    reuse_pre_computed_artifacts: List[str] = [],
+    pipeline_dag_config: Optional[AirflowDagConfig] = {},
 ) -> Path:
     """
     Writes the pipeline job to a path on disk.
 
-    :param artifacts: List of artifact names to be included in the pipeline.
-    :param framework: 'AIRFLOW' or 'SCRIPT'. Defaults to 'SCRIPT' if not specified.
-    :param pipeline_name: Name of the pipeline.
-    :param dependencies: Task dependencies in graphlib format, e.g., {'B':{'A','C'}}
+    Parameters
+    ----------
+    artifacts: List[str]
+        Names of artifacts to be included in the pipeline.
+
+    framework: str
+        "AIRFLOW" or "SCRIPT". Defaults to "SCRIPT" if not specified.
+
+    pipeline_name: Optional[str]
+        Name of the pipeline.
+
+    dependencies: TaskGraphEdge
+        Task dependencies in graphlib format, e.g., ``{"B": {"A", "C"}}``
         means task A and C are prerequisites for task B.
-    :param output_dir: Directory path to save DAG and other pipeline files.
-    :return: Directory path where DAG and other pipeline files are saved.
+
+    output_dir: str
+        Directory path to save DAG and other pipeline files.
+
+    input_parameters: List[str]
+        Names of variables to be used as parameters in the pipeline.
+        Currently, it only accepts variables from literal assignment
+        such as ``a = '123'``. For each variable to be parametrized,
+        there should be only one literal assignment across all
+        artifact code for the pipeline. For instance, if both ``a = '123'``
+        and ``a = 'abc'`` exist in the pipeline's artifact code,
+        we cannot make ``a`` an input parameter since its reference is
+        ambiguous, i.e., we are not sure which literal assignment ``a``
+        refers to.
+
+    reuse_pre_computed_artifacts: List[str]
```

Comment:
same as previous comment

Reply:
Same as the reply to the previous comment.